soh/soh/Enhancements/randomizer/logic.cpp

```diff
@@ -907,7 +907,7 @@ namespace Rando {
     }
 
     bool Logic::CanDetonateUprightBombFlower() {
-        return CanDetonateBombFlowers() || HasItem(RG_GORONS_BRACELET);
+        return CanDetonateBombFlowers() || HasItem(RG_GORONS_BRACELET) || (ctx->GetTrickOption(RT_BLUE_FIRE_MUD_WALLS) && CanUse(RG_BOTTLE_WITH_BLUE_FIRE) && TakeDamage());
```

Comment:
`TakeDamage` has fairy logic which is inaccurate here, since we then must be sure to have 2 bottles, one for blue fire, one for fairy